contracts/src/TeleporterTokenDestination.sol

```diff
@@ -390,9 +403,10 @@ abstract contract TeleporterTokenDestination is
         );
         BridgeMessage memory bridgeMessage = abi.decode(message, (BridgeMessage));
 
-        // If the contract was not previously known to be collateralized, it is now given that
+        // If the contract was not previously known to be registered or collateralized, it is now given that
         // the source has sent a message to mint funds.
-        if (!isCollateralized) {
+        if (!isRegistered || !isCollateralized) {
```

Comment:
was considering whether we should get rid of `isCollateralized` in favor of just `isRegistered`. This wouldn't work for a `NativeTokenDestination` that specifies a zero `intiialReserveBalance` but otherwise think it can

Reply:
I considered it, but it also doesn't make sense for `ERC20Destination`s which start off with `isCollateralized` as `true` from their constructor. Would prefer to leave as is